assets/src/edit-story/components/panels/textStyle/textStyle.js

```diff
@@ -64,9 +64,12 @@ function StylePanel({ selectedElements, pushUpdate }) {
   const textAlign = getCommonValue(selectedElements, 'textAlign');
   const letterSpacing = getCommonValue(selectedElements, 'letterSpacing');
   const lineHeight = getCommonValue(selectedElements, 'lineHeight');
-  const fontStyle = getCommonValue(selectedElements, 'fontStyle');
-  const textDecoration = getCommonValue(selectedElements, 'textDecoration');
   const bold = getCommonValue(selectedElements, 'bold');
+  const hasBold = getCommonValue(selectedElements, 'hasBold');
+  const italic = getCommonValue(selectedElements, 'italic');
+  const hasItalic = getCommonValue(selectedElements, 'hasItalic');
+  const underline = getCommonValue(selectedElements, 'underline');
+  const hasUnderline = getCommonValue(selectedElements, 'hasUnderline');
```

Comment:
Couldn't we just decide if the toggle is on/off based on what's in the content?

We shouldn't track any of these as element attributes IMO. It doesn't make sense to show the toggle on if just one part of the content actually uses it. Tracking all these attributes just for the purpose of showing toggle on/off seems a bit of an overkill.

In display mode, it makes sense to me as a user that if the whole element is wrapped in a tag, only then the toggle is on, otherwise not -- this should be determined based on the content, not the attribute though.
And then, if only parts of the text are marked bold (or something else), then the user can click the Bold toggle which would then apply bold to the whole text element, and then the bold toggle would display as "on".

If the current approach has already been discussed with UX, let me know (a pointer to the discussion would be appreciated).

Thoughts?

Works well in edit mode!

Reply:
Hey, @miina Just to answering your questions:

> Couldn't we just decide if the toggle is on/off based on what's in the content?

I can definitely try an implicit approach, I used these element attributes because I noticed that we need to have different controls while binding and unbinding the styles in the content to make the UI sync with the Design Panel buttons on edit mode when selecting different nodes of the content. Anyway, I will investigate something using your proposal. Thanks!

> In display mode, it makes sense to me as a user that if the whole element is wrapped in a tag, only then the toggle is on, otherwise not -- this should be determined based on the content, not the attribute though.
And then, if only parts of the text are marked bold (or something else), then the user can click the Bold toggle which would then apply bold to the whole text element, and then the bold toggle would display as "on".

Totally agree, my initial focus here was to cover #413 first, but #1007 is partially about what you said. @samitron7 can you please confirm or add some additional UX thoughts here?